### PR TITLE
Dockerimage to install pure::variant as a Capella plugin.

### PR DESCRIPTION
**Pull request by TimoBbz**
This PR introduce a Docker image that allows to create a Capella Container, with
- The following plugins installed:
  - com.ps.consul.eclipse.purevariants.capella.feature.group
  - com.ps.consul.eclipse.purevariants.emf.feature.mapping.feature.group
  - com.ps.consul.eclipse.purevariants.de.enterprise.feature.group
- The injection of the Pure::variant license server with `PUREVARIANT_LICENSE_SERVER` env variable

This is left to do
- [x] Including the pure variant sources (currently they have to live in `the pure-variant/update-site` directory.
- [x] Set the injection of the Pure::variant license server a only one place.
- [x] Complete the _Readme.md_ documentation
- [x] Complete the _Makefile_

### CODE REVIEW

File: pure-variant/startup.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    echo "No scripts are available. Retry in 5 seconds.";
+    sleep 5;
+done
+
+echo "Found the following scripts: $(ls /opt/scripts)"
+sleep 5;
+Xvfb :99 -screen 0 1920x1080x8 -nolisten tcp &
+/opt/capella/capella -data $EASE_WORKSPACE || r=$?
+
+if [[ -n "$r" ]] && [[ "$r" == 158 || "$r" == 0 ]]; then exit 0; else exit 1; fi
```
Comment (MoritzWeber0) [inline pure-variant/startup.sh:20]: You can remove this file :)

Comment (MoritzWeber0) [reply pure-variant/startup.sh:20]: It should be sufficient to have to have it in the `remote` startup.

Comment (TimoBbz) [reply pure-variant/startup.sh]: Done

File: pure-variants/Dockerfile [hunk 2]
```diff
@@ -15,8 +15,9 @@ ONBUILD COPY ./dependencies /tmp/dependencies
 FROM build_${BUILD_TYPE}
 
 USER root
-COPY ./updateSite /tmp/pure-variants
 
+COPY ./updateSite /tmp/pure-variants
+COPY ./license.lic /home/techuser/pure-variants-5/de.license
```
Comment (MoritzWeber0) [inline pure-variants/Dockerfile:20]: The license file should be mounted during runtime, so this line is not needed. 
Do you know if there are different files stored in the `pure-variants-5` directory? In Kubernetes, secrets are mounted as directory (which means that the `/home/techuser/pure-variants-5/de.license` directory is not writable when mounted as read-only). 

I'd prefer a solution where we mount the file to `/inputs/pure-variants/license.lic` and it's copied in the workspace preparation.

File: README.md [hunk 3]
```diff
@@ -453,6 +500,16 @@ connection profile.
 
 We also plan to integrate "dynamic resizing" in the near future.
 
+### Pure::variants
+
+To run the `pure-variants` images, please follow the instructions to run the [`Capella in a remote container`](#capella-in-a-remote-container)
+or [`T4C client in a remote container`](#t4c-client-in-a-remote-container) instructions.
+
+You just have to do changes:
```
Comment (MoritzWeber0) [inline README.md:508]:
```suggestion
You just have to do some changes:
```

File: README.md [hunk 4]
```diff
@@ -453,6 +500,16 @@ connection profile.
...
+
+- Add the environment variable `$PURE_VARIANTS_LICENSE_SERVER` to the `docker run` command.
+- Replace the image name `$BASE/remote` with `$BASE/remote/pure-variants`
+
```
Comment (MoritzWeber0) [inline README.md:512]: Add a section with the volume mount of the license file here.

File: remote/startup.sh [hunk 5]
```diff
@@ -16,6 +16,7 @@ else
 fi
 
 python3 /opt/setup_workspace.py;
+stat /opt/setup_workspace_purevariants.py &>/dev/null && python3 /opt/setup_workspace_purevariants.py
```
Comment (MoritzWeber0) [inline remote/startup.sh:19]: Please add this in the `Dockerfile` of `pure-variants` (e.g. add it in line 5 of the startup.sh).

File: remote/setup_workspace.py [hunk 6]
```diff
@@ -22,7 +22,9 @@
 )
 
 
-def replace_config(path: pathlib.Path, key: str, value: str) -> None:
+def replace_config(
+    logger: logging.Logger, path: pathlib.Path, key: str, value: str
```
Comment (MoritzWeber0) [inline remote/setup_workspace.py:26]: When `LOGGER` is global, you can use it directly. No need to pass it to the method.

File: remote/setup_workspace.py [hunk 7]
```diff
@@ -11,7 +11,7 @@
 from collections import Counter
 
 logging.basicConfig(level=logging.DEBUG)
-LOGGER = logging.getLogger("prepare_workspace")
+LOGGER = logging.getLogger("remote")
```
Comment (MoritzWeber0) [inline remote/setup_workspace.py:14]:
```suggestion
LOGGER = logging.getLogger("TeamForCapella preparation")
```

File: pure-variants/setup_workspace_purevariants.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import re
+
+logging.basicConfig(level=logging.DEBUG)
+LOGGER = logging.getLogger("pure::variants")
```
Comment (MoritzWeber0) [inline pure-variants/setup_workspace_purevariants.py:12]:
```suggestion
LOGGER = logging.getLogger("pure::variants preparation")
```

File: remote/startup.sh (hunk 5 above)
Comment (MoritzWeber0) [inline remote/startup.sh:19]:
```suggestion
python3 /opt/setup_workspace_purevariants.py
```

What's the reason of the `stat` command?

File: pure-variants/setup_workspace_purevariants.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    """This will replace the existing config or add the config (if it doesn't exist)"""
+    path.parent.mkdir(exist_ok=True, parents=True)
+    if path.exists():
+        file_content = path.read_text()
+    else:
+        file_content = ""
+
+    pattern = f"{key}=.+"
+    match = re.search(pattern, file_content)
+    if match:
```
Comment (MoritzWeber0) [inline pure-variants/setup_workspace_purevariants.py:31]:
```suggestion
    if re.search(pattern, file_content):
```

File: pure-variants/setup_workspace_purevariants.py (hunk 8 above)
Comment (TimoBbz) [reply pure-variants/setup_workspace_purevariants.py:12]: This comes from when I tried to isolate the function in its own file for reusability. Fixed.

File: remote/setup_workspace.py (hunk 6 above)
Comment (TimoBbz) [reply remote/setup_workspace.py]: This comes from when I tried to isolate the function in its own file for reusability. Fixed.

File: remote/startup.sh (hunk 5 above)
Comment (TimoBbz) [reply remote/startup.sh:19]: The `startup.sh` file is in the remote so it should not be dependant on if the file provided by the base image exists, hence the `stat`.

File: pure-variants/Dockerfile (hunk 2 above)
Comment (TimoBbz) [reply pure-variants/Dockerfile:20]: The `/home/techuser/pure-variants-5/` directory needs to stay writable, so I implemented your suggestion.

File: remote/startup.sh (hunk 5 above)
Comment (TimoBbz) [reply remote/startup.sh:19]: I think that using `stat` to check if the file exists before executing it is better (more easily debbugable) than injecting the instruction from the Dockerfile with a `sed` command. However, I can switch to this approach if you prefer.

**Pull request merged**